### ISSUE TITLE
Gradient

### ISSUE THREAD

**Issue by Avishaidev** [crisbeto/angular-svg-round-progressbar#121]
Please can you explain how to create gradient as at demo?
Thanks

**Comment by crisbeto** [comment]
[You can check out the demo's source code.](https://github.com/crisbeto/angular-svg-round-progressbar/blob/master/demo/demo.html) You're supposed to put an SVG element, containing the gradient, somewhere on the page:

```html
<svg>
  <linearGradient id="your-gradient-id" x1="0" x2="0" y1="0" y2="1">
    <stop offset="5%"  stop-color="green"/>
    <stop offset="95%" stop-color="gold"/>
  </linearGradient>
</svg>
```

Afterwards you pass the gradient id to the progress circle:

```html
    <round-progress
      [current]="current"
      [max]="max"
      [color]="gradient ? 'url(#gradient)' : color"></round-progress>
```

**Comment by sejr** [comment]
Has anyone had issues when trying to implement this dynamically? I have a bunch of pre-defined gradients and I am trying to switch between them. For some reason, only the first gradient is working.

**Comment by crisbeto** [comment]
Do the gradients have unique ids?

**Comment by sejr** [comment]
@crisbeto Yes they do. I am going to try one possible work-around; if that doesn't work I can provide an update with a JSFiddle or something.

**Comment by mmgmail** [comment]
I found the solution very simply
just leave the color empty like this
`<round-progress
        class="progress__circle2"
        max="100"
        current="$ctrl.carbs"
        color=""
        bgcolor="#f4f4f4"
        radius="30"
        stroke="4"
        rounded="true"
        clockwise="true"
        responsive="true"
        duration="800"
        animation="easeInOutQuart"
        animation-delay="0">
        </round-progress>`

In the global html, add a gradient

`<svg width="0px" height="0px" style="position:absolute;left: -9999px;">
    <defs>
      <linearGradient id="progressCircle1" x1="0" x2="0" y1="0" y2="1">
          <stop offset="0%"  stop-color="#39eeff"/>
          <stop offset="100%"  stop-color="#4da0ff"/>
      </linearGradient>
    </defs>
    <defs>
      <linearGradient id="progressCircle2" x1="0" x2="0" y1="0" y2="1">
          <stop offset="0%"  stop-color="#4af073"/>
          <stop offset="100%"  stop-color="#98e1e7"/>
      </linearGradient>
    </defs>
    <defs>
      <linearGradient id="progressCircle3" x1="0" x2="0" y1="0" y2="1">
          <stop offset="0%"  stop-color="#f0ee49"/>
          <stop offset="100%"  stop-color="#4df078"/>
      </linearGradient>
    </defs>
  </svg>`

In styles write just url (id)

`.round-progress path
    stroke: url(#progressCircle1);    
 }`

**Comment by paresh2190** [comment]
@crisbeto Thanks! Works For me

**Comment by diangogav** [comment]
@crisbeto Thanks!

**Comment by carlosminatoro** [comment]
@mmgmail thanks!, works for me

**Comment by hscheuerle** [comment]
It seems like rerenders on some web browsers get annoyed that ids are being redefined, so going to try putting the definition globally like in the replies here.
